src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -130,16 +88,22 @@ def install_packages(
     which are available on the Snowflake Anaconda channel. These will have
     been deleted from the local packages folder.
     """
-    with Venv() as v:
-        if file_name is not None:
-            pip_install_result = v.pip_install(file_name, RequirementType.FILE)
-            dependencies = v.get_package_dependencies(file_name, RequirementType.FILE)
-
-        if package_name is not None:
-            pip_install_result = v.pip_install(package_name, RequirementType.PACKAGE)
-            dependencies = v.get_package_dependencies(
-                package_name, RequirementType.PACKAGE
-            )
+    if file_name and package_name:
+        raise ClickException(
+            "Could not use package name and requirements file simultaneously"
+        )
+
+    if file_name and not Path(file_name).exists():
+        raise ClickException(f"File {file_name} does not exists.")
+
+    with Venv() as v, NamedTemporaryFile(mode="w") as tmp_requirements:
+        if package_name:
+            tmp_requirements.write(str(package_name))
+            tmp_requirements.flush()
+            file_name = tmp_requirements.name
+
+        pip_install_result = v.pip_install(file_name)
```

Comment:
Simplifies the logic by creating temp requirements.txt for single package. Later on we can move this logic even earlier (for example to a command level)